### PR TITLE
Blog theme - Add images for social sharing

### PR DESCRIPTION
**Pull request by laurieontech**
#22615  Description

This PR adds support for social card images to the blog theme and blog theme core.

## Documentation
Updates to both READMEs included.

Addresses #23640

### CODE REVIEW

File: packages/gatsby-theme-blog/src/components/seo.js [hunk 1]
```diff
@@ -19,13 +19,17 @@ function SEO({ description, lang, meta, keywords = [], title }) {
...
+            siteURL
           }
         }
       }
     `
   )
 
   const metaDescription = description || site.siteMetadata.description
+  const image = imageSource
+    ? `${site.siteMetadata.siteURL}${imageSource}`
```
Comment (marcysutton) [inline packages/gatsby-theme-blog/src/components/seo.js:31]: Handy! Sorta dumb question: does this approach account for pathPrefix / assetPrefix?

Comment (laurieontech) [reply packages/gatsby-theme-blog/src/components/seo.js:31]: Not dumb. My instinct is that it doesn't and shouldn't need to? But that's a good point and I should test it out.

**Comment by laurieontech** [comment]
This should account for testing instructions: https://www.gatsbyjs.org/contributing/code-contributions/#official-theme-development

File: packages/gatsby-theme-blog/src/components/seo.js (hunk 1 above)
Comment (laurieontech) [reply packages/gatsby-theme-blog/src/components/seo.js:31]: Looked into this more. From what I can see querying it's not necessary to account for it specifically.

**Pull request merged**